internal/app/mistapiconnector/mistapiconnector_app.go

```diff
@@ -571,8 +585,8 @@ func (mc *mac) putEtcdKeys(playbackID string, kvs ...string) error {
 	if !resp.Succeeded {
 		panic("unexpected")
 	}
-	glog.Infof("for playbackID=%s created %d keys in ETCD revision=%d", playbackID, len(kvs)/2, resp.Header.Revision)
-	mc.etcdPub2rev[playbackID] = resp.Header.Revision
+	glog.Infof("for playbackID=%s created %d keys in etcd revision=%d", playbackID, len(kvs)/2, resp.Header.Revision)
+	mc.etcdPub2rev[playbackID] = etcdRevData{resp.Header.Revision, kvs}
```

Comment:
Another thing that came to my mind: What happens if we get a write failure above and never get to save the `etcd` entries in memory here? Does the actual stream on Mist fail or does it start without the etcd entries? If the latter, should we "backup" the etcd entries in `etcdPub2rev` first thing in this function and here only add the revision number? (properly handling an invalid revision number like `-1` wherever we use it, like in the delete below)

Reply:
If `putEtcdKeys` call fails then stream will be denied.
BTW, I've spotted small bug - please add fix for it 😄 
```go
		if err != nil {
			glog.Errorf("Error creating ETCD Traefik rule for playbackID=%s streamID=%s err=%v", stream.PlaybackID, stream.ID, err)
			w.WriteHeader(http.StatusInternalServerError)
			w.Write([]byte("false")) // here we tell Mist to deny stream.
// fixme: forgot to return here
		}
	}
	w.Write([]byte(responseURL))
	metrics.StartStream()
	glog.Infof("Responded with '%s'", responseURL)
```

Reply:
[Fixed!](https://github.com/livepeer/stream-tester/pull/90/commits/fd32d4ff96d82fca2622e43f745969fa36a490ec)